python/pdstools/app/datamart_utils.py

```diff
@@ -48,15 +43,21 @@ def import_data(params, default=0, **kwargs):
             )
             params["kwargs"] = ADMDatamart_options()
             if st.checkbox("Use this data source"):
-                if model_file is not None:
+                if predictor_file is None and model_file is not None:
+                    data = import_datamart(
+                        ADMDatamart,
+                        model_df=model_file,
+                        predictor_df=predictor_file,
+                        predictor_filename=None,
+                        **params["kwargs"],
+                    )
+                elif model_file and predictor_file is not None:
```

Comment:
This looks like a weird expression to me - are you trying to check if 'model_file' is True?